crates/oxc_codegen/src/comment.rs

```diff
@@ -40,54 +40,155 @@ impl<'a> Iterator for LineTerminatorSplitter<'a> {
             return None;
         }
 
-        for (index, &byte) in self.text.as_bytes().iter().enumerate() {
-            match byte {
-                b'\n' => {
-                    // SAFETY: Byte at `index` is `\n`, so `index` and `index + 1` are both UTF-8 char boundaries.
-                    // Therefore, slices up to `index` and from `index + 1` are both valid `&str`s.
-                    unsafe {
-                        let line = self.text.get_unchecked(..index);
-                        self.text = self.text.get_unchecked(index + 1..);
-                        return Some(line);
+        // Line terminators will be very rare in most text. So we try to make the search as quick as possible by:
```

Comment:
I'm not sure if this assumption is correct. Line breaks are fairly common in block comments.